bin/eval_inspec_metrics.py

```diff
@@ -0,0 +1,147 @@
+                true_label_embedding, generated_label_embedding
+            )
+            cos_sim_for_avg_emb = encoder.similarity(
+                avg_embedding, generated_label_embedding
+            )
+            count_cos_sim_for_ground_truth.append(cos_sim_for_ground_truth)
+            count_cos_sim_for_avg_emb.append(cos_sim_for_avg_emb)
+
+            if run is None:
+                continue
+            
+            run[f"{split_name}/generated_label"].append(
+                {
+                    "category": summary,
+                    "generated_label": generated_label,
+                    "cos_sim_for_ground_truth": cos_sim_for_ground_truth,
+                    "cos_sim_for_avg_emb": cos_sim_for_avg_emb,
+                }
+            )
+
+        assert len(count_cos_sim_for_ground_truth) > 0, f"Length of {split_name} is 0"
+        average_cos_sim_for_gt = sum(count_cos_sim_for_ground_truth) / len(
+            count_cos_sim_for_ground_truth
+        )
+        average_cos_sim_for_avg_emb = sum(count_cos_sim_for_avg_emb) / len(
+            count_cos_sim_for_avg_emb
+        )
+        result[split_name]["avg_cos_sim_for_ground_truth"] = average_cos_sim_for_gt
+        result[split_name]["avg_cos_sim_for_avg_emb"] = average_cos_sim_for_avg_emb
+
+        if run is None:
+            continue
+        
+        run[f"{split_name}/avg_cos_sim_for_ground_truth"] = average_cos_sim_for_gt
+        run[f"{split_name}/avg_cos_sim_for_avg_emb"] = average_cos_sim_for_avg_emb
+
+    return result
+
+def read_dataset(path: str, random_state: int = 42) -> ParsedDataset:
+    dataset = pd.read_json(path, lines=True)
+
+    X = dataset["doc"]
+    y = dataset["summary"]
+
+    X_train, X_test_eval, y_train, y_test_eval = train_test_split(X, y, test_size=0.2, random_state=random_state)
+    X_test, X_eval, y_test, y_eval = train_test_split(X_test_eval, y_test_eval, test_size=0.5, random_state=random_state)
+
+    parsed_dataset: ParsedDataset = {
+        "train": (X_train, y_train),
+        "test": (X_test, y_test),
+        "eval": (X_eval, y_eval),
+    }
+
+    return parsed_dataset
+
+
+def main(
+    model_name: str, dataset_path: str, encoder_name: str, *args: Any, **kwargs: Any
+) -> None:
+    if run is None:
+        print("Neptune is not initialized. Skipping logging")
+    else:
+      run["dataset_path"] = dataset_path
+      run["parameters"] = {
+          "model_name": model_name,
+          "encoder_name": encoder_name,
+          "args": args,
+          "kwargs": kwargs,
+      }
+
+    encoder = common_utils.get_encoder(encoder_name)
+    model = common_utils.get_model(model_name, encoder, *args, **kwargs)
+    dataset = read_dataset(dataset_path)
+
+    results = eval_model(model, encoder, dataset)
+
+    print(f"Metrics for {model.name}: ", results)
+
+    if run is None:
+        return
+    
+    run["results"] = results
+    run.stop()
+
+
+if __name__ == "__main__":
+    fire.Fire(main)
```

Comment:
I would merge this code with wiki dataset -> merge eval_inspec_metrics with eval_metrics